monero/seed.py

```diff
@@ -0,0 +1,132 @@
+
+    seed_phrase = "" #13 or 25 word mnemonic word string
+    seed_hex = "" # hexadecimal
+
+    def __init__(self):
+        pass
+
+    def endian_swap(self, word):
+        """Given any string, swap bits and return the result.
+
+        :rtype: str
+        """
+        return "".join([word[i:i+2] for i in [6, 4, 2, 0]])
+
+    def encode_seed(self):
+        """Given a hexadecimal string, return it's mnemonic word representation.
+
+        :rtype: str
+        """
+        assert self.seed_hex, "Seed hex not set"
+        assert len(self.seed_hex) % 8 == 0, "Not valid hexadecimal"
+        out = []
+        for i in range(len(self.seed_hex) // 8):
+            word = self.endian_swap(self.seed_hex[8*i:8*i+8])
+            x = int(word, 16)
+            w1 = x % self.n
+            w2 = (x // self.n + w1) % self.n
+            w3 = (x // self.n // self.n + w2) % self.n
+            out += [self.wordlist[w1], self.wordlist[w2], self.wordlist[w3]]
+        self.seed_phrase = " ".join(out)
+        return self.seed_phrase
+
+    def decode_seed(self):
+        """Given a mnemonic word string, return it's hexadecimal representation.
+
+        :rtype: str
+        """
+        assert self.seed_phrase, "Seed phrase not set"
+        phrase = self.seed_phrase.split(" ")
+        assert len(phrase) > 12, "Not valid mnemonic phrase"
+        out = ""
+        for i in range(len(phrase) // 3):
+            word1, word2, word3 = phrase[3*i:3*i+3]
+            w1 = self.wordlist.index(word1)
+            w2 = self.wordlist.index(word2) % self.n
+            w3 = self.wordlist.index(word3) % self.n
+            x = w1 + self.n *((w2 - w1) % self.n) + self.n * self.n * ((w3 - w2) % self.n)
+            out += self.endian_swap("%08x" % x)
+        self.seed_hex = out
+        return self.seed_hex
+
+    def get_checksum(self):
+        """Given a mnemonic word string, return a string of the computed checksum.
+
+        :rtype: str
+        """
+        assert self.seed_phrase, "Seed phrase not set"
+        phrase = self.seed_phrase.split(" ")
+        assert len(phrase) > 12, "Not valid mnemonic phrase"
+        if len(phrase) > 13:
+            # Standard format
+            phrase = phrase[:24]
+        else:
+            # MyMonero format
+            phrase = phrase[:12]
+        wstr = "".join(word[:3] for word in phrase)
+        z = ((crc32(wstr.encode()) & 0xffffffff) ^ 0xffffffff ) >> 0
+        z2 = ((z ^ 0xffffffff) >> 0) % len(phrase)
+        return phrase[z2]
+
+    def validate_checksum(self):
+        """Given a mnemonic word string, confirm seed checksum (last word) matches the computed checksum.
+
+        :rtype: bool
+        """
+        assert self.seed_phrase, "Seed phrase not set"
+        phrase = self.seed_phrase.split(" ")
+        assert len(phrase) > 12, "Not valid mnemonic phrase"
+        is_match = self.get_checksum() == phrase[-1]
+        return is_match
```

Comment:
This could be done in `__init__()` and throw some exception (`ValueError`?) when checksum verification fails.